Processing_data/Classes/StatsBucket.cs

```diff
@@ -0,0 +1,52 @@
+namespace Processing_data.Classes
+{
+    /// <summary>
+    /// Represents a collection of statistics for a set of data
+    /// </summary>
+    public class StatsBucket : Dictionary<Coordinate, Statistics>
+    {
+        /// <summary>
+        /// Adds a data value to the statistics bucket
+        /// </summary>
+        /// <param name="coordinate"></param>
+        /// <param name="data"></param>
+        public void AddData(Coordinate coordinate, double data)
+        {
+            if (!ContainsKey(coordinate))
+            {
+                Add(coordinate, new Statistics(new List<double> { data }));
+            }
+            else
+            {
+                this[coordinate].AddData(data);
+            }
+        }
+
+        public StatsBucket() : base(new CustomEqualityComparer())
+        {
+        }
+
+        /// <summary>
+        /// Prints the statistics for each coordinate in the bucket
+        /// </summary>
+        public void PrintStats()
+        {
+            foreach (KeyValuePair<Coordinate, Statistics> pair in this)
+            {
+                Console.WriteLine($"Coordinate: {pair.Key.Latitude}, {pair.Key.Longitude}. Data: {pair.Value.GetDataCount()} values.");
+                if (pair.Value.GetDataCount() < 100)
```

Comment:
Actually, I meant that we have a stream of events from all sensors in a random order. This is simulated by the data.json. 
 
Each bucket collects 100 events, as they come from the stream / file. For 500 events there will be 5 buckets.

Reply:
Remove this condition